core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -248,6 +247,8 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
 
     initializeControllerContext()
 
+    val pendingPreferredReplicaElections = fetchPendingPreferredReplicaElections()
+    val (topicsToBeDeleted, topicsIneligibleForDeletion) = fetchTopicDeletionsInProgress(pendingPreferredReplicaElections)
```

Comment:
This PR's refactoring of TopicDeletionManager's construction may have complicated things a bit.

Positioning these lines is a bit tricky. We want to load up topic deletion state into `TopicDeletionManager` prior to calling `PartitionStateMachine.startup`, as `triggerOnlinePartitionStateChange` checks `if !controller.topicDeletionManager.isTopicQueuedUpForDeletion(topicAndPartition.topic)`.